src/pages/CodeHandler.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect } from "react";
+import { Redirect, useParams } from "react-router";
+import { Flex, Box, Button, Heading, Spinner } from "@chakra-ui/core";
+import { Link as ReactLink } from "react-router-dom";
+
+import ErrorMessage from "../components/ErrorMessage";
+
+import { sendCode } from "../utils/query";
+
+const CodeHandler = () => {
+  const [state, setState] = useState({
+    error: null,
+    isLoading: true,
+  });
+  const { service } = useParams();
+  const getQueryParams = () =>
+    window.location.search
+      .replace("?", "")
+      .split("&")
+      .reduce(
+        (r, e) => (
+          (r[e.split("=")[0]] = decodeURIComponent(e.split("=")[1])), r
+        ),
+        {}
+      );
+  const { code } = getQueryParams();
+  //Dummy JWT token
+  const token = "ada214142zxczcz213124241";
+  useEffect(() => {
+    async function postCode() {
+      sendCode(service, code, token)
```

Comment:
Is there a difference between `postCode` and `sendCode`? If not lets just use one of them here

Reply:
Yes, they are two different functions.

Reply:
Can we change the name of the functions, they look a little confusing. `postCode` and `sendCode` sound same when reading. Or I suggest we just use the `sendCode` function because `postCode` is only implementing `sendCode` inside it.

Reply:
AFAIK if you want to use async functions inside of ```useEffect```, you have to do it like this. You can't just call ```sendCode```, React doesn't allow that. And ```postCode``` is only a helper function, you won't see it anywhere else. ```sendCode``` is the main function.

Reply:
But sendCode returns a promise right? There should be no problem if we use it inside useEffect

Reply:
Ok I'll try it